### PR TITLE
Refactor SystemStack function for arm targets.

### PR DESCRIPTION
**Pull request by waj334**
Removing usage of arm.AsmFull in favor of writing inline assembly in C.

### CODE REVIEW

**Comment by waj334** [comment]
@aykevl I'm not sure why this one is broken. The existence of task_stack_cortexm.c seems to upset the compiler during the tests.

File: src/internal/task/task_stack_cortexm.c [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <stdint.h>
```
Comment (aykevl) [inline src/internal/task/task_stack_cortexm.c:2]: I think you need to add build tags here, something like this:

```c
//go:build scheduler.tasks && cortexm
```

**Comment by waj334** [comment]
@aykevl Fixed

**Comment by deadprogram** [comment]
Looks correct to me. @aykevl any last feedback here?

**Comment by aykevl** [comment]
Looks good to me. Might be worth rebasing against the dev branch to make sure it still works with https://github.com/tinygo-org/tinygo/pull/3213, or we could just merge it and hope it doesn't break the test.

**Comment by deadprogram** [comment]
> Might be worth rebasing against the dev branch to make sure it still works with #3213

That is a good idea. Could you please do that @waj334 ?

**Comment by waj334** [comment]
@deadprogram done

**Comment by deadprogram** [comment]
Thank you very much for this @waj334 now merging.

**Pull request merged**